From 34199bdccd2784638a6442b1724edb5f72b4ac02 Mon Sep 17 00:00:00 2001
From: Mark Adler <madler@alumni.caltech.edu>
Date: Sun, 11 Jan 2015 18:46:03 -0800
Subject: [PATCH 1/2] Simplify justname().

This uses strrchr() instead of a manual search, and avoids an
illegal pointer calculation for purists (one less than the start
of the buffer).
---
 pigz.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/pigz.c b/pigz.c
index cc8aaae..1b1d89a 100644
--- a/pigz.c
+++ b/pigz.c
@@ -3231,11 +3231,8 @@ local char *justname(char *path)
 {
     char *p;
 
-    p = path + strlen(path);
-    while (--p >= path)
-        if (*p == '/')
-            break;
-    return p + 1;
+    p = strrchr(path, '/');
+    return p == NULL ? path : p + 1;
 }
 
 /* Copy file attributes, from -> to, as best we can.  This is best effort, so
-- 
2.1.4

